### PR TITLE
List: selectAll should not unselect disabled items (T1050340)

### PR DESCRIPTION
**Pull request by Zedwag**

### CODE REVIEW

File: js/ui/selection/selection.strategy.js [hunk 1]
```diff
@@ -137,19 +137,20 @@ export default Class.inherit({
...
         const dataFilter = this.options.filter();
-        let selectedItems = this.options.selectedItems;
+        let selectedItems = this.options.ignoreDisabledItems ? this.options.selectedItems : this.options.selectedItems.filter(item => !item?.disabled);
 
         if(dataFilter) {
             selectedItems = dataQuery(selectedItems).filter(dataFilter).toArray();
         }
 
         const selectedItemsLength = selectedItems.length;
+        const disabledItemsLength = items.length - this.getSelectableItems(items).length;
```
Comment (ksercs) [inline js/ui/selection/selection.strategy.js:147]: shouldn't it be disabled**Selected**ItemsLength?

File: testing/tests/DevExpress.ui.widgets/listParts/editingTests.js [hunk 2]
```diff
@@ -408,6 +408,24 @@ QUnit.test('selectAll/unselectAll for \'allPages\' selectAllMode', function(asse
     assert.equal(loading.callCount, 2, 'no load during unselect all');
 });
 
+QUnit.test('unselectAll method should not unselect disabled items', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/listParts/editingTests.js]:
```suggestion
QUnit.test('unselectAll method should not unselect disabled items (T1050340)', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets/listParts/editingTests.js [hunk 3]
```diff
@@ -408,6 +408,24 @@ QUnit.test('selectAll/unselectAll for \'allPages\' selectAllMode', function(asse
...
+    const instance = $('#list').dxList({
+        dataSource: items,
+        selectedItemKeys: '1',
+        selectionMode: 'all',
+        keyExpr: 'text',
+    }).dxList('instance');
+
+    instance.selectAll();
+
+    assert.deepEqual(instance.option('selectedItems'), items, 'all item are selected');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/listParts/editingTests.js]: Why all items should be selected here? Shouldn't it be ignored as in the test below?

File: testing/tests/DevExpress.ui.widgets/listParts/editingTests.js (hunk 2 above)
Comment (Zedwag) [reply testing/tests/DevExpress.ui.widgets/listParts/editingTests.js]: fixed

File: testing/tests/DevExpress.ui.widgets/listParts/editingTests.js (hunk 3 above)
Comment (Zedwag) [reply testing/tests/DevExpress.ui.widgets/listParts/editingTests.js]: `disabled` item is selected on init, so `selectAll()` here just selects second item (not disabled).
actually there is no need in this, I have refactored this test - now they are both selected on init and then we call `unselectAll()`

File: js/ui/selection/selection.strategy.js (hunk 1 above)
Comment (Zedwag) [reply js/ui/selection/selection.strategy.js:147]: nope, we expect `true` when all non disabled items are selected
all non disabled items = all items - all disabled items

**Review by ksercs — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**